data/soapvalidator/Mail/Persona/SendOnBehalfOfMailUsingPrimaryAccount.xml

```diff
@@ -0,0 +1,372 @@
+			</t:response>
+		</t:test>
+
+		<!-- verify right using DiscoverRightsRequest -->
+		<t:test>
+			<t:request>
+				<DiscoverRightsRequest xmlns="urn:zimbraAccount">
+					<right>sendOnBehalfOf</right>
+				</DiscoverRightsRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:DiscoverRightsResponse/acct:targets"
+					attr="right" match="sendOnBehalfOf" />
+				<t:select
+					path="//acct:DiscoverRightsResponse/acct:targets/acct:target/acct:email[@addr='${test_account2.name}']" />
+			</t:response>
+		</t:test>
+
+		<!-- Login as gratee test_account1.name -->
+		<t:property name="server.zimbraAccount" value="${test_acct2.server}" />
+		<t:test id="auth_acct1" required="true" depends="acct1_setup">
+			<t:request>
+				<AuthRequest xmlns="urn:zimbraAccount">
+					<account by="name">${test_account2.name}</account>
+					<password>${defaultpassword.value}</password>
+				</AuthRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:AuthResponse/acct:lifetime" match="^\d+$" />
+				<t:select path="//acct:AuthResponse/acct:authToken" set="authToken" />
+			</t:response>
+		</t:test>
+
+		<t:test required="true">
+			<t:request>
+				<GetIdentitiesRequest xmlns="urn:zimbraAccount" />
+			</t:request>
+			<t:response>
+				<t:select path="//acct:GetIdentitiesResponse">
+					<t:select path="//acct:identity[@name='DEFAULT']" attr="id"
+						set="account1.identity.default.id" />
+				</t:select>
+			</t:response>
+		</t:test>
+
+		<t:test required="true">
+			<t:request>
+				<ModifyIdentityRequest xmlns="urn:zimbraAccount">
+					<identity name="DEFAULT">
+						<a name="zimbraPrefFromAddress">${test_account1.name}</a>
+					</identity>
+				</ModifyIdentityRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:ModifyIdentityResponse" />
+			</t:response>
+		</t:test>
+
+		<!-- Send email with From: Primary Account -->
+		<t:test id="sendmail1" required="true">
+			<t:request>
+				<SendMsgRequest xmlns="urn:zimbraMail">
+					<m>
+						<e t="f" a="${test_account2.name}" />
+						<e t="t" a="${test_account3.name}" />
+						<su> ${msg02.subject}</su>
+						<mp ct="text/plain">
+							<content> ${message.content1}</content>
+						</mp>
+					</m>
+				</SendMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//zimbra:Code" match="service.PERM_DENIED"
+					emptyset="1" />
+				<t:select path="//mail:SendMsgResponse/mail:m" attr="id"
+					set="message.id2" />
+			</t:response>
+		</t:test>
+
+		<!-- Verify message details using GetMsgRequest -->
+		<t:test id="GetMsgRequest">
+			<t:request>
+				<GetMsgRequest xmlns="urn:zimbraMail">
+					<m id="${message.id2}" />
+				</GetMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//mail:GetMsgResponse/mail:m[@id='${message.id2}']">
+					<t:select path="//mail:e[@t='f']" attr="a"
+						match="${test_account2.name}" />
+					<t:select path="//mail:e[@t='t']" attr="a"
```

Comment:
Add validation for checking @t='s' element. In case of SOBO, sender value will be updated in this element.

Reply:
Added validation for element @t='s'